backend/api/oauth.py

```
@@ -0,0 +1,204 @@
+        }
+    
+    # Create new user
+    new_user = create_oauth_user(db, user_info)
+    oauth_states.pop(state, None)  # Clean up state
+    
+    return create_user_login_response(new_user)
+
+@router.post("/google/link")
+async def link_google_account(
+    request: AccountLinkingRequest,
+    db: Session = Depends(get_db)
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)